kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/container/orchestration/ContainerNetworkConfiguration.java

```diff
@@ -0,0 +1,95 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Eurotech and/or its affiliates and others
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *  Eurotech
+ *******************************************************************************/
+
+package org.eclipse.kura.container.orchestration;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Objects;
+
+import org.osgi.annotation.versioning.ProviderType;
+
+/**
+ * Object which represents a container network configuration used to when
+ * requesting the generation of a new container instance.
+ *
+ * @noimplement This interface is not intended to be implemented by clients.
+ * @since 2.4
+ *
+ */
+@ProviderType
+public class ContainerNetworkConfiguration {
+
+    private String networkMode;
+
+    private ContainerNetworkConfiguration() {
+    }
+
+    /**
+     * 
+     * returns the network mode a container will be created with. supported modes
+     * include: 'bridge', 'none', 'container:', 'host'.
+     * 
+     * @return
+     */
+    public String getNetworkMode() {
+        return this.networkMode;
+    }
+
+    /**
+     * Creates a builder for creating a new {@link ContainerNetworkConfiguration}
+     * instance.
+     *
+     * @return the builder.
+     */
+    public static ContainerNetworkConfigurationBuilder builder() {
+        return new ContainerNetworkConfigurationBuilder();
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(this.networkMode);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (!(obj instanceof ContainerNetworkConfiguration)) {
+            return false;
+        }
+        ContainerNetworkConfiguration other = (ContainerNetworkConfiguration) obj;
+        return Objects.equals(this.networkMode, other.networkMode);
+    }
+
+    public static final class ContainerNetworkConfigurationBuilder {
+
+        private String networkMode = "bridge";
```

Comment:
I'm not sure that using this default is ok, setting it might introduce a change in behavior since the previous version, which did not configure a network mode explicitly.
If we use it we must be sure that `bridge` is the default for this parameter for all container engines.

Otherwise we could use the empty string as a default value or maybe changing the type of `ContainerNetworkConfiguration.networkMode` to `Optional<String>`, update public APIs accordingly and keeping `Optional.empty()` as default.

A `Optional.empty()` value means that the implementation must not supply the network mode parameter to the container engine, relying on its defaults (the current behavior).